code/game/objects/structures/tables_racks.dm

```diff
@@ -406,7 +406,11 @@
 	. = ..()
 	if(flags & NODECONSTRUCT)
 		return
-	if(!isliving(AM))
+	if(isliving(AM))
+		var/mob/living/M = AM
+		if(M.incorporeal_move || M.flying || M.floating)
+			return
+	else
```

Comment:
You can keep the early return here since it only accepts `/mob/living` anyway.
```suggestion
	if(!isliving(AM))
		return
	var/mob/living/L = AM
	if(L.incorporeal_move || L.flying || L.floating)
		return

```